### PR TITLE
Moved module from puppet-staging to puppet-archive

### PR DESCRIPTION
**Pull request by angrox**
So, again, 2nd try, sorry for the inconvenience. 
This pull request moves the tomcat module from the puppet-staging module to the puppet-archive module.

### CODE REVIEW

File: manifests/extras.pp [hunk 1]
```diff
@@ -6,30 +6,38 @@
...
+    cleanup => false,
     notify  => Service[$::tomcat::service_name_real]
   }
 
-  staging::file {
-    'catalina-jmx-remote.jar':
-      target => "${::tomcat::catalina_home_real}/lib/extras/catalina-jmx-remote-${::tomcat::version_real}.jar",
-      source => "http://archive.apache.org/dist/tomcat/tomcat-${::tomcat::maj_version}/v${::tomcat::version_real}/bin/extras/catalina-jmx-remote.jar"
+  archive {
+    "${::tomcat::catalina_home_real}/lib/extras/catalina-jmx-remote-${::tomcat::version_real}.jar":
```
Comment (antoineco) [inline manifests/extras.pp:23]: Would it be possible to keep the original resource names here?

File: manifests/install/archive.pp [hunk 2]
```diff
@@ -31,21 +38,23 @@
...
+    ensure => directory,
   } ->
-  staging::file { "apache-tomcat-${::tomcat::version_real}.tar.gz": source => $::tomcat::archive_source_real } ->
-  staging::extract { "apache-tomcat-${::tomcat::version_real}.tar.gz":
-    target  => $::tomcat::catalina_home_real,
-    creates => "${::tomcat::catalina_home_real}/bin",
-    user    => $::tomcat::tomcat_user_real,
-    group   => $::tomcat::tomcat_group_real,
-    strip   => 1
+  archive { "${::tomcat::catalina_home_real}/apache-tomcat-${::tomcat::version_real}.tar.gz":
```
Comment (antoineco) [inline manifests/install/archive.pp:44]: Same remark regarding the resource names

File: manifests/instance.pp [hunk 3]
```diff
@@ -559,17 +580,19 @@
...
       alias   => $catalina_home_alias
     } ->
-    staging::file { "apache-tomcat-${version_real}.tar.gz": source => $archive_source_real } ->
-    staging::extract { "apache-tomcat-${version_real}.tar.gz":
-      target  => $catalina_home_real,
-      creates => "${catalina_home_real}/bin",
-      user    => $tomcat_user,
-      group   => $tomcat_group,
-      strip   => 1
+    archive { "${catalina_home_real}/apache-tomcat-${version_real}.tar.gz":
```
Comment (antoineco) [inline manifests/instance.pp:583]: Same remark regarding the resource names

File: manifests/instance.pp [hunk 4]
```diff
@@ -1150,30 +1173,34 @@
...
+        cleanup => false,
         notify  => Service[$service_name_real]
       }
-
-      staging::file {
-        "instance ${name} catalina-jmx-remote.jar":
-          target => "${catalina_base_real}/lib/extras/catalina-jmx-remote-${::tomcat::version_real}.jar",
-          source => "http://archive.apache.org/dist/tomcat/tomcat-${::tomcat::maj_version}/v${::tomcat::version_real}/bin/extras/catalina-jmx-remote.jar"
+      archive {
+        "${catalina_base_real}/lib/extras/catalina-jmx-remote-${::tomcat::version_real}.jar":
```
Comment (antoineco) [inline manifests/instance.pp:1186]: Same remark regarding the resource names

File: manifests/init.pp [hunk 5]
```diff
@@ -267,6 +282,12 @@
   validate_re($service_ensure, '^(stopped|running)$', '$service_ensure must be either \'stopped\', or \'running\'')
   validate_array($listeners, $executors, $connectors, $realms, $valves, $globalnaming_environments, $globalnaming_resources, $context_watchedresources, $context_parameters, $context_environments, $context_listeners, $context_valves, $context_resourcedefs, $context_resourcelinks, $catalina_opts, $java_opts, $jpda_opts)
   validate_hash($server_params, $svc_params, $threadpool_params, $http_params, $ssl_params, $ajp_params, $engine_params, $host_params, $context_params, $context_loader, $context_manager, $context_realm, $context_resources, $custom_variables)
+  validate_bool($checksum_verify)
+  validate_re($checksum_type, '(none|md5|sha1|sha2|sh256|sha384|sha512)', 'The checksum type needs to be one of the following: none|md5|sha1|sha2|sh256|sha384|sha512')
```
Comment (antoineco) [inline manifests/init.pp:286]: Something like `sha2qwertz` or `md5asdfg` would match here, unless you surround the regexp with `^$` as above

File: manifests/instance.pp [hunk 6]
```diff
@@ -245,6 +260,12 @@
   validate_re($service_ensure, '^(stopped|running)$', '$service_ensure must be either \'stopped\', or \'running\'')
   validate_array($listeners, $executors, $connectors, $realms, $valves, $globalnaming_environments, $globalnaming_resources, $context_watchedresources, $context_parameters, $context_environments, $context_listeners, $context_valves, $context_resourcedefs, $context_resourcelinks, $catalina_opts, $java_opts, $jpda_opts)
   validate_hash($server_params, $svc_params, $threadpool_params, $http_params, $ssl_params, $ajp_params, $engine_params, $host_params, $context_params, $context_loader, $context_manager, $context_realm, $context_resources, $custom_variables)
+  validate_bool($checksum_verify)
+  validate_re($checksum_type, '(none|md5|sha1|sha2|sh256|sha384|sha512)', 'The checksum type needs to be one of the following: none|md5|sha1|sha2|sh256|sha384|sha512')
```
Comment (antoineco) [inline manifests/instance.pp:264]: Same remark regarding the regexp

**Comment by antoineco** [comment]
I'm very sorry @angrox for the delay :pray: I've reviewed and tested your commit and I'm all up for merging. I still have a couple of minor remarks regarding the resource names and regexp matching.

Do you think you'd have time to review that soonish? If not I can also do it.

Thank you again for your work :clap:

File: manifests/extras.pp (hunk 1 above)
Comment (angrox) [inline manifests/extras.pp:23]: Of course, I just changed it to 100% unique

**Comment by angrox** [comment]
@antoineco NP - this happens to me too :-) Feel free to modify the regexp and variable names to fit in the module. If you wish I can do that also.

**Comment by antoineco** [comment]
Merged on https://github.com/antoineco/aco-tomcat/commits/puppet-archive
I will apply some minor changes discussed here and merge on master.

**Comment by angrox** [comment]
Thanks antoineco. I have been on another project and had no time making the changes! Thanks a lot!

**Comment by antoineco** [comment]
No problem, I had some bug fixes on my list as well so I did all the modifications in one go :) 
Your changes made it to v1.4.0, which was pushed yesterday :+1:

**Pull request closed without merging**